calculator/main.py

```diff
@@ -26,7 +26,20 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if operation not in calc["operations"]:
+        raise InvalidOperation('"{}" operation not supported.'.format(operation))
+    
+    
+    if params is None or not all([isinstance(param, (float,int)) and type(param) is not bool for param in params]):
+        raise InvalidParams("Given params are invalid.")
+        
```

Comment:
I think the part that was tripping us up, was that initially if we test 2 bool values it'd deny it, but after a while when we finished it started accepting bools as valid params instead. We were trying to avoid that, so we ended up with this large chunk of a line.